File: LES_SERIE_S1_IA_FS_ALGO_PYTHON_C/Série_de_TD-TP_3/exercice10.py
chaine=""
i=1
while True:
    c=str(input(f"saisir le caractere numero {i}: "))[0]
    if c==".":
        break
    elif c=="0":
        chaine=chaine+" "
    else:
        i+=1
        chaine=chaine+c
print(chaine)

File: LES_SERIE_S1_IA_FS_ALGO_PYTHON_C/Série_de_TD-TP_3/exercice10.c
#include <stdio.h>
#include <string.h>
int i;
char c,chaine[100];
int main(){
    i=0;
    do{
        i++;
        printf("saisir le caractere %d : ",i);
        scanf(" %c",&c);
        if(c!='.')
        {
            int len=strlen(chaine);
            if(c=='0')
            {
                chaine[len]= ' ';
                chaine[len+1]= '\0';

            }
            else
            {
                chaine[len]=c;
                chaine[len+1]='\0';
            }
        }
    }while(c!='.');
    printf("%s\n",chaine);
}